ports/rp2/machine_uart.c

```diff
@@ -169,6 +252,36 @@ STATIC mp_obj_t machine_uart_make_new(const mp_obj_type_t *type, size_t n_args,
         self->invert = args[ARG_invert].u_int;
     }
 
+    self->read_lock = false;
+
+    // Set the buffer size and allocate the buffer
+    size_t rxbuf_len = DEFAULT_BUFFER_SIZE;
+    // get the length if set
+    if (args[ARG_rxbuf].u_int > 0) {
+        rxbuf_len = args[ARG_rxbuf].u_int;
+        // minimal size is DEFAULT_BUFFER_SIZE
+        // maximal size is MAX_BUFFER_SIZE
+        if (rxbuf_len < DEFAULT_BUFFER_SIZE) {
+            rxbuf_len = DEFAULT_BUFFER_SIZE;
+        } else if (rxbuf_len > MAX_BUFFER_SIZE) {
+            mp_raise_ValueError(MP_ERROR_TEXT("rxbuf too large"));
+        }
+    }
+
+    // Set the buffer size and allocate the buffer
+    size_t txbuf_len = DEFAULT_BUFFER_SIZE;
+    // get the length if set
+    if (args[ARG_txbuf].u_int > 0) {
+        txbuf_len = args[ARG_txbuf].u_int;
+        // minimal size is DEFAULT_BUFFER_SIZE
+        // maximal size is MAX_BUFFER_SIZE
+        if (txbuf_len < DEFAULT_BUFFER_SIZE) {
+            txbuf_len = DEFAULT_BUFFER_SIZE;
```

Comment:
As above, I think the check for minimum can be much less, like 2.

Reply:
As above.